nipype/pipeline/plugins/multiproc.py

```diff
@@ -178,44 +193,54 @@ def _report_crash(self, node, result=None):
             return report_crash(node)
 
     def _clear_task(self, taskid):
-        del self._taskresult[taskid]
+        del self._task_obj[taskid]
 
     def _submit_job(self, node, updatehash=False):
         self._taskid += 1
         if hasattr(node.inputs, 'terminal_output'):
             if node.inputs.terminal_output == 'stream':
                 node.inputs.terminal_output = 'allatonce'
 
-        self._taskresult[self._taskid] = \
+        self._task_obj[self._taskid] = \
             self.pool.apply_async(run_node,
-                                  (node, updatehash),
-                                  callback=release_lock)
+                                  (node, updatehash, self._taskid),
+                                  callback=self._async_callback)
         return self._taskid
 
+    def _close(self):
+        self.pool.close()
+        return True
+
     def _send_procs_to_workers(self, updatehash=False, graph=None):
         """ Sends jobs to workers when system resources are available.
             Check memory (gb) and cores usage before running jobs.
         """
         executing_now = []
 
         # Check to see if a job is available
-        jobids = np.flatnonzero((self.proc_pending == True) & \
+        currently_running_jobids = np.flatnonzero((self.proc_pending == True) & \
```

Comment:
that should be revised.

the following line sets a flag when a job is submitted.

https://github.com/nipy/nipype/blob/a746633d72d3e861a4b1a52038dd40a590792e24/nipype/pipeline/plugins/base.py#L371